libs/search-widget/src/components/answer/Answer.svelte

```diff
@@ -20,14 +21,65 @@
   $: notEnoughData = text === 'Not enough data to answer this.';
   $: isMobile = isMobileViewport(innerWidth);
 
-  const sources = answer.sources?.resources ? getSortedResults(Object.values(answer.sources?.resources)) : [];
+  const sources = answer.sources?.resources
+    ? getSortedResults(removeNotCitedContent(Object.values(answer.sources?.resources), answer.citations))
+    : [];
+
+  function removeNotCitedContent(resources: Search.FindResource[], citations?: Citations) {
+    if (!citations || Object.keys(citations).length === 0) {
+      return resources;
+    }
+    const citedParagraphs = Object.keys(citations);
+    const citedResources = citedParagraphs.reduce((acc, paragraphId) => {
+      const resId = paragraphId.split('/')[0];
+      if (!acc.includes(resId)) {
+        acc.push(resId);
+      }
+      return acc;
+    }, [] as string[]);
+
+    return (
+      resources
+        // Remove resources that are not cited
+        .filter((resource) => citedResources.includes(resource.id))
+        // Remove paragraphs that are not cited
+        .map((resource) => ({
+          ...resource,
+          fields: Object.entries(resource.fields).reduce(
+            (acc, field) => {
+              acc[field[0]] = {
+                ...field[1],
+                paragraphs: Object.entries(field[1].paragraphs).reduce(
+                  (acc, paragraph) => {
+                    if (citedParagraphs.includes(paragraph[0])) {
+                      acc[paragraph[0]] = paragraph[1];
+                    }
+                    return acc;
+                  },
```

Comment:
1. you should avoid shadow variables: having an `acc` inside another `acc` is not a good practice

2. it would be easier to read if instead of keeping `field` and `paragraph` as array containing the id and the actual field and paragraph, you did something like 
```ts
(acc, [fieldId, field]) => {
```
 and 
```ts
(acc, [paragraphId, paragraph]) => {
```